pipe.go

```diff
@@ -63,8 +63,8 @@ var _ wire = (*pipe)(nil)
 
 type pipe struct {
 	conn            net.Conn
-	clhks           atomic.Value // closed hook, invoked after the conn is closed
-	pshks           atomic.Value // pubsub hook, registered by the SetPubSubHooks
+	clhks           atomic.Value          // closed hook, invoked after the conn is closed
+	pshks           atomic.Pointer[pshks] // pubsub hook, registered by the SetPubSubHooks
```

Comment:
Hi @nithinputhenveettil,

Could you also sync the change to https://github.com/redis/rueidis/blob/51c2ebfd3f20c292150905b91644ea96295bff23/mock/result.go#L190 to fix the CI failures?